### PR TITLE
Admin components: Add default links to logo footers (Jetpack's and Automattic's)

### LINKED ISSUE #23108: My Jetpack: Footer Jetpack link links to nothing. Make it link to jetpack.com
image

### LINKED ISSUE #23112: Components: AdminPage -  Make Automattic logo on the footer default to link to automattic.com

### PR DESCRIPTION
**Pull request by leogermani**
Fixes #23108
Fixes #23112

#### Changes proposed in this Pull Request:

* Add new prop to add a link to the Module name in the JetpackFooter defaulting to `https://jetpack.com`
* Add link to jetpack.com in My Jetpack footer
* Makes the A8C logo default to linking to `https://automattic.com`
* Updates components to deal with the new defaults.

#### Jetpack product discussion

NA

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Open Storybook
* Go to Js packages > Components > Jetpack Footer
* Check there's a new `moduleNameHref` prop in the Contorls tab
* Make sure no `a` tag is rendered in the module name when this prop is empty
* Make sure a `a` tag is properly rendered when there is a value
* Make sure styles look good
Screen Shot 2022-03-03 at 9 29 12 AM

Now on My Jetpack
* Visit My jetpack
* make sure the "Jetpack" in the footer links to jetpack.com
* open the interstitial pages and check the same thing

Note 1: I didn't make this link the default because jetpack.com is the default link to the a8c logo already (shouldn't it be a8c.com??)

### CODE REVIEW

File: projects/js-packages/components/components/jetpack-footer/style.scss [hunk 1]
```diff
@@ -7,6 +7,11 @@
 	max-width: 1128px;
 
 	color: #000;
+
+	a, a:hover, a:visited {
+		color: #000;
```
Comment (retrofox) [inline projects/js-packages/components/components/jetpack-footer/style.scss:12]: probably we should try to use CSS vars here

**Review by retrofox — DISMISSED** [review]
LGTM

**Review by oskosk — COMMENTED** [review]
I think that if we go this way of enabling such a possibility as to change what we put besides the Jetpack logo and where it links to, the footer logo component should default to linking to jetpack.com (prop `moduleNameHref` should default to `https://jetpack.com`) and be an External Link

**Comment by oskosk** [comment]
Pushed dcbb5615c68c77809265c8711b4e8bf8b7cff73b, 7e77b0c781707d240a2ca9180d92aa006fd3bc79 and 56b5d23198df1d4871793a8e6ae8b44feb7ef352 so now this PR also addresses #23112

**Review by oskosk — APPROVED** [review]
LGTM

**Pull request merged**